docs/how-to-contribute/index.md

```diff
@@ -0,0 +1,268 @@
+- `npm test-lib`: runs the lib's unit tests in headless mode (no browser UI).
+- `npm test-lib-local`: runs the lib's unit test with the browser UI (for local development of the lib).
+- `npm run webdriver-update`: installs Chrome webdriver in the required version for the E2E tests.
+- `npm run e2e`: runs the E2E tests from the demo application, see section [Unit Tests and E2E Tests](#unit-tests-and-e2e-tests).
+
+### Building
+
+- `npm run build-lib`: builds the lib in productive mode
+- `npm run build-lib-dev`: builds the lib in development mode
+- `npm build-app`: builds the demo application in productive mode
+- `npm run yalc-publish-lib`: builds the lib and publishes it locally using [yalc](https://www.npmjs.com/package/yalc).
+
+## Library Build Options
+
+The library is built in productive mode for publication, using [tsconfig.lib.prod.json](https://github.com/dasch-swiss/dsp-ui-lib/blob/main/projects/dsp-ui/tsconfig.lib.prod.json).
+
+To perform additional checks during development, the library can be built in development mode,
+using [tsconfig.lib.json](https://github.com/dasch-swiss/dsp-ui-lib/blob/main/projects/dsp-ui/tsconfig.lib.json). Development builds use Ivy and perform [strict template checks](https://angular.io/guide/template-typecheck#template-type-checking).
+
+## Unit Tests and E2E Tests
+
+There are unit tests for the lib (`projects/dsp-ui/src`)
+and some basic unit tests for the demo application playground components (`src/app`).
+
+The E2E tests have to be run from the demo application.
+Unlike the unit tests, they require a running instance of DSP-API compatible with the release specified in [vars.mk](https://github.com/dasch-swiss/dsp-ui-lib/blob/main/vars.mk): {% include-markdown "../../vars.mk" %}
+Before running the E2E tests, reload the data in DSP-API's db.
+The E2E tests can be run as follows:
+
+- `npm run webdriver-update`
+- `npm run e2e`
+
+## DSP-UI library developement
+
+### Adding New Components etc. to the Library
+
+Install [Angular CLI](https://angular.io/cli) globally:
+
+```shell
+ npm install -g @angular/cli
+```
+
+To add additional components etc. to a module of the library, run:
+
+```shell
+ng generate component [path/in/the/module/][name-of-component] --project @dasch-swiss/dsp-ui
+```
+
+For example, to create a new component `test` in the action module, run:
+
+```shell
+ng generate component action/test --project @dasch-swiss/dsp-ui
+```
+
+This command will create a folder called `test` inside of `projects/dsp-ui/src/lib/action/` containing the component's class, template, style, and spec files.
+
+Before testing the new component inside of the demo app, you have to rebuild it after each change:
+
+```shell
+npm run build-lib
+```
+
+Alternatively, you can use [Live Reloading](#run-the-application-locally-in-dev-mode-with-live-reloading).
+
+## Demo Application (Playground)
+
+### Adding New Component to the Demo Application
```

Comment:
why having all the words in camelcase in all the titles?

Reply:
I suggest to do these changes in a separate refactor task. We have to rewrite and update the "how-to-use"-documentation anyway.